salt/runners/net.py

```diff
@@ -224,8 +224,19 @@ def _find_interfaces_mac(ip):  # pylint: disable=invalid-name
         if not device_ipaddrs.get('result', False):
             continue
         for interface, interface_ipaddrs in six.iteritems(device_ipaddrs.get('out', {})):
-            ip_addresses = interface_ipaddrs.get('ipv4', {}).keys()
-            ip_addresses.extend(interface_ipaddrs.get('ipv6', {}).keys())
+            ip_addresses = set()
+            try:
+                ip_addresses.add(list(interface_ipaddrs.get('ipv4', {}).keys())[0])
+            except IndexError:
+                pass
+            try:
+                ip_addresses.add(list(interface_ipaddrs.get('ipv4', {}).keys())[1])
+            except IndexError:
+                pass
+            try:
+                ip_addresses.add(list(interface_ipaddrs.get('ipv6', {}).keys())[0])
+            except (AttributeError, IndexError):
+                pass
```

Comment:
A cleaner way to switch to a `set` would be: 
```
ip_addresses = set(interface_ipaddrs.get('ipv4', {}).keys())
ip_addresses.update(set(interface_ipaddrs.get('ipv6', {}).keys()))
```

Reply:
Yes, I completely agree. Thanks!